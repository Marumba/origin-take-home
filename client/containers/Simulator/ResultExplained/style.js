import styled from '@emotion/styled';

import * as S from '../shared/style.shared';

export const ResultExplained = styled(S.ContentHolder)`
	${({ theme }) => theme.mixins.caption}

	background: ${({ theme }) => theme.colors.blueGray10};

	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;
