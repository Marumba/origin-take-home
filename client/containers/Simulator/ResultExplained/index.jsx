import React from 'react';
import PropTypes from 'prop-types';

import * as S from './style';

function ResultExplained({ customCss }) {
	return (
		<S.ResultExplained as="p" data-testid="simulator-result-explained" customCss={customCss}>
			Testing the text <strong>amount</strong> an the line height
		</S.ResultExplained>
	);
}

ResultExplained.propTypes = {
	customCss: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({ styles: PropTypes.string })])
};

ResultExplained.defaultProps = {
	customCss: undefined
};

export default ResultExplained;
