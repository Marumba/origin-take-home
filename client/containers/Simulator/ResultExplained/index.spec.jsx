import React from 'react';

import { render, screen } from '~/helpers/testing.customRtl';

import ResultExplained from '.';

describe('Container Simulator ResultExplained', () => {
	it('should render without any props given', () => {
		const { container } = render(<ResultExplained />);
		expect(screen.getByTestId('simulator-result-explained')).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should incorporate de given css', () => {
		render(
			<ResultExplained
				customCss={{
					background: 'red'
				}}
			/>
		);
		expect(screen.getByTestId('simulator-result-explained')).toHaveStyleRule('background', 'red');
	});
});
