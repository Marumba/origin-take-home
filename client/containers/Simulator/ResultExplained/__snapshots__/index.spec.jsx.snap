// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Simulator ResultExplained should render without any props given 1`] = `
.emotion-0 {
  padding: 24px;
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 16px;
  background: #F4F8FA;
}

.emotion-0.highEmphasis {
  color: #1E2A32;
}

.emotion-0 strong {
  font-weight: 600;
}

<p
  class="emotion-0 emotion-1"
  data-testid="simulator-result-explained"
>
  Testing the text 
  <strong>
    amount
  </strong>
   an the line height
</p>
`;
